test/test_integration.py

```diff
@@ -401,6 +401,86 @@
     'expect_out': ['aa,bb,c_c', '1,2,3', '4,5,6']
 }
 
+underscore_csv = {
+    'name': 'underscore_csv',
+    'raw_data': ['a,b,c',
+                 '1,2,y',
+                 '4,5,2_0'],
+    'script': {"name": "underscore_csv",
+               "resources": [
+                   {"dialect": {"do_not_bulk_insert": "True"},
+                    "name": "underscore_csv",
+                    "schema": {
+                        "fields": [
+                            {
+                                "name": "record_id",
+                                "type": "pk-auto"
+                            },
+                            {
+                                "name": "a",
+                                "type": "int"
+                            },
+                            {
+                                "name": "b",
+                                "type": "int"
+                            },
+                            {
+                                "name": "c",
+                                "type": "char"
+                            }
+                        ]
+                    },
+                    "url": "http://example.com/underscore_csv.txt"
+                    }
+               ],
+               "retriever": "True",
+               "retriever_minimum_version": "2.0.dev",
+               "version": "1.0.0",
+               "urls": {"underscore_csv": "http://example.com/underscore_csv.txt"}
+               },
+    'expect_out': ['record_id,a,b,c', '1,1,2,y', '2,4,5,2_0']
+}
+
+underscore_int_csv = {
+    'name': 'underscore_int_csv',
+    'raw_data': ['a,b,c',
+                 '1,2,3',
+                 '4,5,20'],
```

Comment:
Were you thinking that the last value here should be `2_0` instead of `20`? If so I'm wondering whether or not we really want to read in a `2_0` from a csv file as `20`. I don't think in most cases that will be the intended value.

Reply:
With the way things are moving, a new pep like "2_0", we shall def see files with such data in the near feature. I just wanted to be sure that we are handling both.

Reply:
But I could remove that test in case we think it may not fit for now

Reply:
While I see Python folks using that notation in code I don't see it becoming standard usage in csv's since Python is the only language that has introduced that notation. Personally I feel like we should be treating `2_0` as a string regardless, but maybe this is a larger discussion that deserve's its own issue. So, let's remove that test for now, merge this, and open an issue to discuss what to do with that notation.

Reply:
Sounds perfect